### PR TITLE
Sample course: add sample comments for some feedback responses #5485

### LINKED ISSUE #5485: Sample course: add sample comments for some feedback responses
EDITED (by damith)
The sample course (that we create when we add a new instructor account) currently has some comments on students.
Suggested:  Add some sample comments for feedback responses too

### PR DESCRIPTION
**Pull request by Mynk96**
Fixes #5485

### CODE REVIEW

**Comment by Mynk96** [comment]
samplecomments
Ui changes..
For now, I have added only sample comments, If there are any particular comments please let me know.
Also, I am not able to figure out the convention for giving names to feedbackResponseComment objects
i.e `comment1FromT1C1ToR1Q1S1C1`(what are T,C,R,Q,S,C) in it. So for that, I have just incremented the counter for now (1->2->3).

**Comment by Mynk96** [comment]
Ready For Review!

File: src/main/resources/InstructorSampleData.json [hunk 1]
```diff
@@ -2764,23 +2764,85 @@
...
+	  "lastEditedAt": "2017-06-03 9:20 PM +0530"
+	},
+	"comment3FromT3C3ToR3Q3S3C3": {
+	  "courseId": "demo.course",
+	  "feedbackSessionName": "Session with different question types",
+	  "feedbackQuestionId": "5",
+	  "giverEmail": "teammates.demo.instructor@demo.course",
+	  "giverSection": "Tutorial Group 1",
+	  "receiverSection": "Tutorial Group 1",
+	  "feedbackResponseId": "5%alice.b.tmms@gmail.tmt%francis.g.tmms@gmail.tmt",
```
Comment (chowyb) [inline src/main/resources/InstructorSampleData.json]: Francis Gabriel is actually in Tutorial Group 2, not Tutorial Group 1.

**Comment by Mynk96** [comment]
@chowyb I have made the changes and also added one more comment for tutorial 2., but I don't why the build is getting failed please review!

**Comment by chowyb** [comment]
@Mynk96 the current failure may be due to an unstable test that is currently being fixed (#7452), and should therefore work properly once that PR is merged into master, and this branch is updated. In the meantime, ignore that particular test failure for now.

File: src/main/resources/InstructorSampleData.json [hunk 2]
```diff
@@ -2763,24 +2763,112 @@
...
+	  "lastEditedAt": "2017-06-03 9:23 PM +0530"
+	},
+	"comment1FromT1C1ToR2Q2S1C1": {
+	  "courseId": "demo.course",
+	  "feedbackSessionName": "Session with different question types",
+	  "feedbackQuestionId": "2",
+	  "giverEmail": "teammates.demo.instructor@demo.course",
+	  "giverSection": "Tutorial Group 2",
+	  "receiverSection": "Tutorial Group 2",
+	  "feedbackResponseId": "2%charlie.d.tmms@gmail.tmt%charlie.d.tmms@gmail.tmt",
```
Comment (chowyb) [inline src/main/resources/InstructorSampleData.json]: Charlie Davis is in Tutorial Group 1.

Comment (Mynk96) [inline src/main/resources/InstructorSampleData.json]: @chowyb 
`"charlie.d.tmms@demo.course": {`
      `"googleId": "charlie.d.tmms.sampleData",`
      `"email": "charlie.d.tmms@gmail.tmt",`
      `"course": "demo.course",`
      `"name": "Charlie Davis",`
      `"comments": "This student's name is Charlie Davis",`
      `"team": "Team 2",`
      `"section": "Tutorial Group 2"`

**Comment by Mynk96** [comment]
@chowyb Ready for Review!

File: src/main/resources/InstructorSampleData.json (hunk 2 above)
Comment (chowyb) [inline src/main/resources/InstructorSampleData.json]: Hmm, looks like I remembered the default data wrongly. I always thought names A to D were in Team 1 =P

**Comment by Mynk96** [comment]
@chowyb did you create a new Instructor?

**Comment by chowyb** [comment]
@Mynk96 oh right, I thought you hadn't fixed the issue you posted on the issue yet =P guess that part doesn't need to be traced then =P seems to work fine on the new instructor.

**Comment by Mynk96** [comment]
@chowyb should a add a newline at the end of the file? The previous InstructorSampleData.JSON didn't have any.

**Comment by chowyb** [comment]
@Mynk96 depends on whether #7434 is merged in before this one. Adding a newline should probably be left to that PR, so if that PR's merged in already then add a newline, otherwise leave it as it is for now.

**Comment by Mynk96** [comment]
@chowyb Ready for Review!

File: src/main/resources/InstructorSampleData.json [hunk 3]
```diff
@@ -2763,23 +2763,111 @@
...
+      "showGiverNameTo": [
+        "GIVER",
+        "RECEIVER",
+        "RECEIVER_TEAM_MEMBERS",
+        "INSTRUCTORS"
       ],
-      "showGiverNameTo": [],
       "isVisibilityFollowingFeedbackQuestion": false,
-      "createdAt": "2016-03-01 11:59 PM UTC",
+      "createdAt": "2017-06-03 9:17 PM +0530",
```
Comment (chowyb) [inline src/main/resources/InstructorSampleData.json]: Missed this - maybe we should leave all times at UTC (the course is already at UTC).

Comment (Mynk96) [inline src/main/resources/InstructorSampleData.json]: oh sorry!

**Comment by Mynk96** [comment]
@chowyb changes done!

**Comment by damithc** [comment]
Given that no tests broke, I assume this area is not covered in any tests. We should cover the sample course in tests because if it gets messed up, new users get a bad first impression. Can be a separate issue. May be @sujeet14108 is interested :-p

**Comment by damithc** [comment]
> Also, I am not able to figure out the convention for giving names to feedbackResponseComment objects
> i.e comment1FromT1C1ToR1Q1S1C1(what are T,C,R,Q,S,C) in it. So for that, I have just incremented the counter for now (1->2->3).

@Mynk96 Did you figure this one out?

**Comment by Mynk96** [comment]
@damithc Yes, @chowyb helped me in understanding the convention.

**Comment by sujeet14108** [comment]
> Given that no tests broke, I assume this area is not covered in any tests. We should cover the sample course in tests because if it gets messed up, new users get a bad first impression. Can be a separate issue. May be @sujeet14108 is interested :-p

@damithc will look at it

**Comment by sujeet14108** [comment]
@damithc  As i have seen there is tests for successful creation of demo course and for generating demo course id in `AdminInstructorAccountAddActionTest` . But the details inside the course are not checked.
So should we create test for that also ?

**Comment by damithc** [comment]
> @damithc As i have seen there is tests for successful creation of demo course and for generating demo course id in AdminInstructorAccountAddActionTest . But the details inside the course are not checked.
> So should we create test for that also ?

Yes @sujeet14108 , we can look into enhancing the existing test to cover content of the sample course.

**Comment by damithc** [comment]
@Mynk96 can you confirm that this feature is working in the latest master branch?

**Comment by Mynk96** [comment]
@damithc yes the sample comments are there in the feedback session.

**Comment by damithc** [comment]
> @damithc yes the sample comments are there in the feedback session. But the issue #7481 exists.

Thanks @Mynk96 . Apparently I was looking in the wrong session. While we are at it, why not add a few more comments in the `First team feedback session` too? That's the one users are more likely to look at.

**Comment by Mynk96** [comment]
Okay, @damithc I'll create a new issue and will add few more comments in the `First team feedback session`.

**Comment by LiHaoTan** [comment]
@Mynk96 just to close the issue I went to check mine again, and this time it showed up. Sorry, I made a mistake too.

Apparently I checked out the wrong branch when attempting to test this,
I checked out your branch of `5654-Instructor-view-student-record-allow-editing/deleting-of-feedback-Response-Comments` instead of `5485-Sample-course-add-sample-comments-for-some-feedback-responses`. Apparently both your branches had the same feedback response comment, but the wrong branch had `comment1FromT1C1ToR1Q1S1C1` instead of `comment1FromT1C1ToR1Q2S1C1`.

**Pull request merged**